### PR TITLE
fix(server): follow symlinks when zipping assets

### LINKED ISSUE #9335: Symlinked photos in external library download as a zip of symlinks
### The bug

I have some photos that are presented to immich in an "external library".  That external library happens to be a git-annex repository, so all of the photos are actually symlinks to somewhere in its `.git` directory.  Immich displays these files with no problems.  However, if one tries to download a set of them, one ends up with a zip file full of symlinks and no actual images.

For example, here's what I get from unzipping a zip that immich generated that contained two images:

```
 ls -l
total 8
lrwxrwxrwx. 1 rob rob 207 May  8 20:17 IMG_6611.JPG -> ../../../.git/annex/objects/kp/74/SHA256E-s6007094--24cb5836a3634c8c663ea53846cf05d8c1471dbfbe6d61e346db0c944d002e42.JPG/SHA256E-s6007094--24cb5836a3634c8c663ea53846cf05d8c1471dbfbe6d61e346db0c944d002e42.JPG
lrwxrwxrwx. 1 rob rob 207 May  8 20:17 IMG_6613.JPG -> ../../../.git/annex/objects/vx/jw/SHA256E-s5979551--0895fe458d242ba5ea5fbefc3e7816c7621ed54e0bd8f06471f2431f30e64b5c.JPG/SHA256E-s5979551--0895fe458d242ba5ea5fbefc3e7816c7621ed54e0bd8f06471f2431f30e64b5c.JPG
```

As you can see, both of those files are symlinks, and not actually image files.

Downloading the images individually works fine.

### The OS that Immich Server is running on

nixos

### Version of Immich Server

1.103.1

### Version of Immich Mobile App

n/a

### Platform with the issue

- [ ] Server
- [X] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
n/a: Not using docker-compose -- immich is running in podman.
```

### Your .env content

```Shell
n/a
```

### Reproduction steps

```bash
1. Have an external library that has images that are symlinks to somewhere internally within the library.
2. Select two of those images in the immich web interface
3. Observe that the resulting zip file does not actually contain the images
...
```

### Relevant log output

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by C-Otto**
fixes #9335

### CODE REVIEW

File: server/src/services/download.service.ts [hunk 1]
```diff
@@ -83,10 +84,10 @@ export class DownloadService {
         filename = `${parsedFilename.name}+${count}${parsedFilename.ext}`;
       }
 
-      zip.addFile(originalPath, filename);
+      promises.push(zip.addFile(originalPath, filename));
```
Comment (jrasm91) [inline server/src/services/download.service.ts:87]: Why don't you just get the real path here and revert every other change?

**Comment by C-Otto** [comment]
This works fine on my instance, but I'm unsure of how the test code works on filesystems that don't support symlinks (Windows?). Furthermore, as `git annex` protects files by marking them as read-only, the corresponding attributes are also used for the ZIP file returned by immich. It'd be a bit more convenient to have those files with `rw-r-r` flags instead - but that might be an unrelated fix.

**Review by jrasm91 — APPROVED** [review]
I changed the `realpath` to a repo call, which made the test a lot easier to write. I also added `mode: 644`, which sets the mode on the file in the archive. Do you want to re-test it on your library?

**Comment by C-Otto** [comment]
Thanks, that works like a charm! :rocket:

**Pull request merged**